packages/api-server/src/convert-tx.ts

```diff
@@ -160,7 +160,7 @@ export function buildPolyjuiceArgs(
       toAddressWhenNativeTransfer.slice(2),
       "hex"
     );
-    toAddressBuf.copy(argsBuf, 52 + (data.length - 2));
+    toAddressBuf.copy(argsBuf, 52 + dataBuf.byteLength);
```

Comment:
To avoid this kind of bug, how about adding some logs/assertions for `argsBuf.size != 52 + dataBuf.byteLength`?

Reply:
good idea, I will try adding some checks

Reply:
the args length is already allocated with right size. check is no need. bug comes from mismatch the copy position, leave a todo comment to add unit test for this functions